src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -1599,14 +1627,43 @@ private void WindowClosing(object sender, CancelEventArgs e)
         }
 
         /// <summary>
-        /// Saves the state of currently displayed extension windows. This is needed because the closing event is
+        /// Saves the state of currently displayed extension windows and tabs. This is needed because the closing event is
         /// not called on child windows: https://docs.microsoft.com/en-us/dotnet/api/system.windows.window.closing
         /// </summary>
-        private void SaveExtensionWindowsState()
+        private void SaveExtensionsState()
         {
+            //loop over all active extension windows and tabs.
             foreach (var window in ExtensionWindows.Values)
             {
                 SaveExtensionWindowSettings(window);
+                SaveExtensionOpenState(window);
+            }
+            //for any new extensions that are opened for the first time
+            foreach (var tab in ExtensionTabItems)
+            {
+                SaveExtensionOpenState(tab);
+            }
+            //update open state of all existing view extension in setting, if option to remember extensions is enabled in preferences
+            if (dynamoViewModel.PreferenceSettings.EnablePersistExtensions) {
+                var settings = dynamoViewModel.PreferenceSettings.ViewExtensionSettingsList;
```

Comment:
maybe just always do this - and just let the preference control loading?
If I am not mistaken you are checking the setting here and at load time?

Reply:
Yes, at load time to display extensions and here to update them.
Can you elaborate on `always do this`?

Reply:
always update them, no matter what state the preference is in.

Reply:
So basically remove the `EnablePersistExtensions` property check?

Reply:
yeah, just an idea to simplify it - is it needed?

Reply:
not needed, will update, won't hurt to default to false.